test/spmd/tensor/test_matrix_ops.py

```
@@ -127,6 +131,150 @@ def test_t(self):
+            new_dt = torch.bmm(input_dt, mat_2_dt)
+            self.assertEqual(new_dt.to_local(), local_result)
+
+
+        assertionErrorFormat = \
+            "Test failed! AssertionError: Result tensors not equal! Spec: {0} {1}."
+        runtimeErrorFormat = \
+            "Test failed! RuntimeError: \"{2}\"! Spec: {0} {1}."
+        shard0_spec = Shard(0)
+        shard1_spec = Shard(1)
+        shard2_spec = Shard(2)
+        shard_specs = [shard0_spec, shard1_spec, shard2_spec]
+        shard_specs_comb = itertools.product(shard_specs, shard_specs)
+        passlist = [
+            [shard0_spec, shard0_spec],
+            [shard2_spec, shard1_spec],
+        ]
+
+        def placementEqual(
+            p1: Placement, p2: Placement) -> bool:
+            if (p1.is_shard() and p2.is_shard()):
+                return p1.dim == p2.dim
+            elif (p1.is_replicate() and p2.is_replicate()):
+                return True
+            elif (p1.is_partial() and p2.is_partial()):
+                return True
+            else:
+                return False
+
+        def specEqual(
+            s1: Sequence[Placement], s2: Sequence[Placement]) -> bool:
+            if (len(s1) != len(s2)):
+                return False
+            else:
+                boolSeq = map(lambda t: placementEqual(t[0], t[1]), zip(s1, s2))
+                return functools.reduce(lambda x, y: x and y, boolSeq, True)
+
+        def specInSeq(
+            s: Sequence[Placement], l: Sequence[Sequence[Placement]]) -> bool:
+            boolSeq = map(lambda r: specEqual(s, r), l)
+            return functools.reduce(lambda x, y: x or y, boolSeq, False)
+
+        # tests that currently pass
+        for spec in passlist:
+            test_placement_comb([spec[0]], [spec[1]])
+
+        # TODO: support these tests
+        shard_specs_comb = [spec for spec in shard_specs_comb if not specInSeq(spec, passlist)]
+        for spec in shard_specs_comb:
+            try:
```

can we just do `self.assertRaise(Exception)` instead of print the exception to stdout? that way we can capture the failure in test when it silently pass later.